packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx

```diff
@@ -5,7 +5,7 @@ import { ComboboxActionProps } from "../../Combobox.types";
 import { ComboboxContext } from "../../ComboboxProvider";
 
 export function ComboboxAction(props: ComboboxActionProps) {
-  const { searchValue } = useContext(ComboboxContext);
+  const { searchValue, setOpen } = useContext(ComboboxContext);
```

Comment:
wait, I totally forgot - we need to use `handleClose` not just `setOpen` when we are closing the combobox. 

```
const handleClose = useCallback(() => {
    setOpen(false);
    setSearchValue("");

    onSearch && onSearch("");
    onClose && onClose();

    if (selectedOptions.length > 0) {
      shouldScroll.current = true;
    }

```
- sets the open state to false
- clears search
- calls callbacks for a cleared search & on close
- scroll state management

if we don't do all that, when it is defaulting to close on click you'll see things like your previous search still being present when you open it up again.

not great that we didn't have a test for that. we should add one!

so it would need to be something like


```
const { searchValue, handleClose } = useContext(ComboboxContext);
...
const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
  if (!props.keepOpenOnClick) {
    handleClose()
  }

  props.onClick(e, options);
};
```

one thing I'd wanna make sure of is if `!props.keepOpenOnClick` is the safest way to do that. since it's optional but without a literal code implemented default it could be
- `undefined` which gives us `true`
- `false` which gives us `true`
- `true` which gives us `false`

I think that's all correct?